python/ray/data/dataset.py

```diff
@@ -1336,10 +1337,14 @@ def filter(
     def repartition(
         self,
         num_blocks: int,
+        max_num_rows_per_block: Optional[int] = None,
```

Comment:
I think this name might confuse people. IIUC if this is `10` and there are 12 rows per block, you'd end up with 12 rows in each output block and exceed the specified `max_num_rows_per_block`.

Maybe `target_num_rows_per_block` would be a more accurate name, and we should also explain that this might not be exactly adhered to.